_posts/en/2022-01-26-build-website-with-nextjs.md

```diff
@@ -0,0 +1,365 @@
+
+To start, delete the index.js content, to start from a blank page and not with the NextJS welcome page.
+
+Now, we need to build our website layout, let's create a reusable React component and integrate it in the global NextJS App component to see it in each generated pages.
+
+### Construct a layout
+
+First, create a components folder in the `src` folder:
+
+```shell
+mkdir src/components
+```
+
+Create the Layout component folder tree:
+
+```shell
+mkdir src/components/Layout
+touch src/components/Layout/Layout.js
+touch src/components/index.js
+touch src/components/Layout.module.css
+```
+
+* Let's add the Layout HTML code and add some styling to it:
+
+`src/components/Layout.js`
+```js
+// Libs
+import PropTypes from "prop-types"
+import Head from "next/head";
+import { useRouter } from "next/router";
+
+// Components
+import { FlexContainer, ORIENTATION } from "../FlexContainer";
+import { ElevenLabsLogo } from "../ElevenLabsLogo/ElevenLabsLogo";
+import { SpaceCraftButton } from "../SpaceCraftButton";
+
+// CSS Module
+import styles from "./Layout.module.css";
+import Image from "next/image";
+
+export function Layout({ children: pageContent }) {
+  const router = useRouter()
+  const isHomePage = router.pathname === "/"
+
+  const handleContactClick = (e) => {
+    e.preventDefault()
+    router.push("/contact")
+  }
+
+    return <div className={styles["main-container"]}>
+    <Head>
+      <title>AstroTeams</title>
+      <meta name="description" content="Eleven Labs Astro Teams" />
+      <link rel="icon" href="/favicon.ico" />
+    </Head>
+
+    <main className={styles.main}>
+      <FlexContainer orientation={ORIENTATION.VERTICAL}>
+        <ElevenLabsLogo />
+        { pageContent }
+      </FlexContainer>
+      { isHomePage && <SpaceCraftButton onClick={handleContactClick}>Contact</SpaceCraftButton> }
+    </main>
+
+    <footer className={styles.footer}>
+      <Image
+        alt="Eleven Labs Logo"
+        src="/logo-eleven-labs-large.png"
+        objectFit="contain"
+        width="150px"
+        height="50px"
+        priority
+      />
+      <div className={styles["footer__label"]}>NextJS Demo App</div>
+    </footer>
+  </div>
+}
+
+Layout.propTypes = {
+  /**
+   * Page content
+  */
+  children: PropTypes.node.isRequired,
+}
+```
+
+> **Note**: don't forget to add the styling in the CSS module. Also, export your component in the index.js.
+
+Now, add the component to the `src/pages/_app.js`:
+
+```js
+// Components
+import { Layout } from "../components/Layout"
+
+// Global CSS
+import "../styles/globals.css"
+
+function MyApp({ Component, pageProps }) {
+  return <Layout><Component {...pageProps} /></Layout>
+}
+
+export default MyApp
+```
+
+Now, each page generated from the pages folder, add the Layout, we can see it in our index page:
+
+![astro teams layout](assets/2022-01-26-build-website-with-nextjs/website-layout.png)
+
+Now, just add the HTML content for the index page.
```

Comment:
I had the feeling that I had to do this by myself, but that what you will explain in next part, am I wrong? :) 

```suggestion
Now, we will add the HTML content for the index page.
```